ksql-api/src/main/java/io/confluent/ksql/api/server/ServerUtils.java

```diff
@@ -32,14 +37,29 @@ private ServerUtils() {
 
   public static void handleError(final HttpServerResponse response, final int statusCode,
       final int errorCode, final String errMsg) {
-    final JsonObject errResponse = new JsonObject().put("status", "error")
+    final JsonObject errResponse = createErrResponse(errorCode, errMsg);
+    response.setStatusCode(statusCode).end(errResponse.toBuffer());
+  }
+
+  public static JsonObject createErrResponse(final int errorCode, final String errMsg) {
+    return new JsonObject().put("status", "error")
         .put("errorCode", errorCode)
         .put("message", errMsg);
-    response.setStatusCode(statusCode).end(errResponse.toBuffer());
   }
 
   public static void unhandledExceptonHandler(final Throwable t) {
     log.error("Unhandled exception", t);
   }
 
+  public static JsonObject decodeJsonObject(final Buffer buffer,
+      final RoutingContext routingContext) {
+    try {
+      return new JsonObject(buffer);
+    } catch (DecodeException e) {
+      handleError(routingContext.response(), 400, ERROR_CODE_INVALID_JSON,
+          "Invalid JSON in request args");
+      return null;
```

Comment:
can we have this return `Optional` to indicate that this can return null?

Reply:
This is by design. I prefer it returning null as it makes the code that uses it simpler. If we return Optional then the using code needs to call get() each time to get the actual object which is more verbose. But I appreciate this is a contentious point.

Reply:
fair enough - I can see both sides

Reply:
I'm not sure I understand why the other is better than:
```
final JsonObject requestBody = decodeJsonObject(routingContext.getBody(), routingContext);
    if (requestBody == null) {
      return;
    }
```
It seems to be doing the exact thing, but with more code?

Reply:
Changed in later PR